terraformtest.go

```diff
@@ -0,0 +1,74 @@
+package terraformtest
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/tidwall/gjson"
+)
+
+type TFPlan struct {
+	Data []byte
+}
+
+type TFResource struct {
+	Metadata map[string]string
+	Total    int
+	Values   map[string]string
+}
+
+func Equal(tfResource TFResource, tfPlan TFPlan) bool {
+	rootModuleResources := gjson.GetBytes(tfPlan.Data, "planned_values.root_module.resources")
+	childModulesResources := gjson.GetBytes(tfPlan.Data, "planned_values.root_module.child_modules.#.resources")
+	if tfResource.Total > 0 {
+		if tfResource.Total != len(rootModuleResources.Array())+len(childModulesResources.Array()) {
+			return false
+		}
+	}
+
+	for k, v := range tfResource.Metadata {
+		value := gjson.GetBytes([]byte(rootModuleResources.Raw), fmt.Sprintf("0.0.%s", k))
+		if !value.Exists() {
+			value = gjson.GetBytes([]byte(childModulesResources.Raw), fmt.Sprintf("0.0.%s", k))
+			if !value.Exists() {
+				return false
```

Comment:
We could do with more helpful test output, couldn't we? For example, if a certain attribute or resource is missing from the plan, or different to what we specified, we'd like to see that information.

Reply:
I was following the initial idea of acting like the cmp.Diff